waku/v2/node/peer_manager/peer_manager.nim

```diff
@@ -99,49 +94,55 @@ proc loadFromStorage(pm: PeerManager) =
       # Do not manage self
       return
 
-    pm.peerStore.addressBook[peerId] = storedInfo.addrs
-    pm.peerStore.protoBook[peerId] = storedInfo.protos
-    pm.peerStore.keyBook[peerId] = storedInfo.publicKey
-    pm.peerStore.connectionBook[peerId] = NotConnected  # Reset connectedness state
-    pm.peerStore.disconnectBook[peerId] = disconnectTime
-  
+    # nim-libp2p books
+    pm.peerStore[AddressBook][peerId] = storedInfo.addrs
+    pm.peerStore[ProtoBook][peerId] = storedInfo.protos
+    pm.peerStore[KeyBook][peerId] = storedInfo.publicKey
+    pm.peerStore[AgentBook][peerId] = storedInfo.agent
+    pm.peerStore[ProtoVersionBook][peerId] = storedInfo.protoVersion
+
+    # custom books
+    pm.peerStore[ConnectionBook][peerId] = NotConnected  # Reset connectedness state
+    pm.peerStore[DisconnectBook][peerId] = disconnectTime
+    pm.peerStore[SourceBook][peerId] = storedInfo.origin
+
   let res = pm.storage.getAll(onData)
   if res.isErr:
     warn "failed to load peers from storage", err = res.error
     waku_peers_errors.inc(labelValues = ["storage_load_failure"])
   else:
     debug "successfully queried peer storage"
-  
+
 ##################
 # Initialisation #
-##################   
+##################
 
 proc onConnEvent(pm: PeerManager, peerId: PeerID, event: ConnEvent) {.async.} =
-  if not pm.peerStore.addressBook.contains(peerId):
+  if not pm.peerStore[AddressBook].contains(peerId):
     ## We only consider connection events if we
     ## already track some addresses for this peer
     return
 
   case event.kind
   of ConnEventKind.Connected:
-    pm.peerStore.connectionBook[peerId] = Connected
+    pm.peerStore[ConnectionBook][peerId] = Connected
     if not pm.storage.isNil:
       pm.storage.insertOrReplace(peerId, pm.peerStore.get(peerId), Connected)
     return
   of ConnEventKind.Disconnected:
-    pm.peerStore.connectionBook[peerId] = CanConnect
+    pm.peerStore[ConnectionBook][peerId] = CanConnect
     if not pm.storage.isNil:
       pm.storage.insertOrReplace(peerId, pm.peerStore.get(peerId), CanConnect, getTime().toUnix)
     return
 
 proc new*(T: type PeerManager, switch: Switch, storage: PeerStorage = nil): PeerManager =
   let pm = PeerManager(switch: switch,
-                       peerStore: WakuPeerStore.new(),
+                       peerStore: switch.peerStore,
```

Comment:
`peerStorage` and `peerStore` names are confusing. Is there any plan to rename any of them?